crates/ark/src/lsp/diagnostics.rs

```diff
@@ -171,42 +171,52 @@ fn is_within_magrittr_pipe(node: Node, context: &mut DiagnosticContext) -> bool
     }
 }
 
-fn handle_dotty_assignment(lhs: Node, context: &mut DiagnosticContext) -> Result<()> {
+fn handle_dotty_assignment(lhs: Node, context: &mut DiagnosticContext) -> Option<()> {
     // Check that the lhs is a subset call of the form `.[]`.
-    lhs.is_subset().into_result()?;
-    let function = lhs.child_by_field_name("function").into_result()?;
-    function.is_identifier_or_string().into_result()?;
-    let value = context.contents.node_slice(&function)?.to_string();
-    (value == ".").into_result()?;
+    if !lhs.is_subset() {
+        return None;
+    };
+
+    let function = lhs.child_by_field_name("function")?;
+    if !function.is_identifier_or_string() {
+        return None;
+    };
+
+    let value = context.contents.node_slice(&function).ok()?.to_string();
```

Comment:
I have a slight preference for actually logging the potential error here and manually doing `return None`, like in an `Err` match arm

Reply:
Ah I see how often you use it below though, hmm.